File: app/src/main/java/com/example/geofence20/LocationAlertIntentService.java
package com.example.geofence20;

import android.app.IntentService;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;

import com.google.android.gms.location.Geofence;
import com.google.android.gms.location.GeofencingEvent;

public class LocationAlertIntentService extends IntentService {

	private static final String IDENTIFIER = "LocationAlertIS";

	public LocationAlertIntentService() {
		super(IDENTIFIER);
	}

	@Override
	protected void onHandleIntent(Intent intent) {
		GeofencingEvent geofencingEvent = GeofencingEvent.fromIntent(intent);

		if (geofencingEvent.hasError()) {
			Log.e(IDENTIFIER, "erro");
		}

		String requestId = geofencingEvent.getTriggeringGeofences().get(0).getRequestId();

		Log.i(requestId, geofencingEvent.toString());

		int geofenceTransition = geofencingEvent.getGeofenceTransition();

		if (geofenceTransition == Geofence.GEOFENCE_TRANSITION_ENTER ||
				geofenceTransition == Geofence.GEOFENCE_TRANSITION_DWELL) {
			//setar no firebase
			Toast.makeText(getApplicationContext(), "Entrou na cerca", Toast.LENGTH_LONG).show();
		} else if (geofenceTransition == Geofence.GEOFENCE_TRANSITION_EXIT) {
			//setar no firebase
			Toast.makeText(getApplicationContext(), "Saiu da cerca", Toast.LENGTH_LONG).show();
		}
	}
}

File: app/src/main/java/com/example/geofence20/LocationAlertIntentService.kt
package com.example.geofence20

import android.app.IntentService
import android.content.Intent
import android.util.Log
import android.widget.Toast

import com.google.android.gms.location.Geofence
import com.google.android.gms.location.GeofencingEvent
import com.google.firebase.database.FirebaseDatabase
import android.support.v4.content.LocalBroadcastManager



class LocationAlertIntentService : IntentService(IDENTIFIER) {

	companion object {
		private val IDENTIFIER = "LocationAlertIS"
	}

	override fun onHandleIntent(intent: Intent?) {
		val geofencingEvent = GeofencingEvent.fromIntent(intent)

		if (geofencingEvent.hasError()) {
			Log.e(IDENTIFIER, "erro")
		}

		val requestId = geofencingEvent.triggeringGeofences[0].requestId

		Log.i(requestId, geofencingEvent.toString())

		val geofenceTransition = geofencingEvent.geofenceTransition

		val db = FirebaseDatabase.getInstance().reference

		if (geofenceTransition == Geofence.GEOFENCE_TRANSITION_ENTER || geofenceTransition == Geofence.GEOFENCE_TRANSITION_DWELL) {
			db.child(requestId).setValue(1)
			Toast.makeText(applicationContext, "Entrou na cerca", Toast.LENGTH_LONG).show()
			if (requestId == "validacaoCasa") {
				db.child("casaDigital").setValue(0)
				val msg = Intent(MainActivity.FILTRO_KEY)
				msg.putExtra(MainActivity.MENSAGEM_KEY, true)
				LocalBroadcastManager.getInstance(this).sendBroadcast(msg)
			}
		} else if (geofenceTransition == Geofence.GEOFENCE_TRANSITION_EXIT) {
			db.child(requestId).setValue(0)
			Toast.makeText(applicationContext, "Saiu da cerca", Toast.LENGTH_LONG).show()
		}
	}

}
